processor/memorylimiterprocessor/memorylimiter_test.go

```diff
@@ -440,3 +108,278 @@ func TestNoDataLoss(t *testing.T) {
 	err = processor.Shutdown(context.Background())
 	require.NoError(t, err)
 }
+
+// TestMetricsMemoryPressureResponse manipulates results from querying memory and
+// check expected side effects.
+func TestMetricsMemoryPressureResponse(t *testing.T) {
+	md := pmetric.NewMetrics()
+	ctx := context.Background()
+
+	tests := []struct {
+		name           string
+		mlCfg          *Config
+		getMemFn       func() (uint64, error)
+		readMemStatsFn func(m *runtime.MemStats)
+		expectError    bool
+	}{
+		{
+			name: "fixed_limit",
+			mlCfg: &Config{
+				CheckInterval:       1 * time.Nanosecond,
+				MemoryLimitMiB:      1024,
+				MemorySpikeLimitMiB: 512,
+			},
+			expectError: false,
+		},
+		{
+			name: "percent_limit",
+			mlCfg: &Config{
+				CheckInterval:         1 * time.Nanosecond,
+				MemoryLimitPercentage: 50,
+				MemorySpikePercentage: 30,
+			},
+			expectError: false,
+		},
+		{
+			name: "fixed_limit",
+			mlCfg: &Config{
+				CheckInterval:         1 * time.Nanosecond,
+				MemoryLimitPercentage: 50,
+				MemorySpikePercentage: 30,
+			},
+			getMemFn:       totalMemory,
+			readMemStatsFn: readMemStats,
+			expectError:    true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.getMemFn != nil {
+				memorylimiter.GetMemoryFn = tt.getMemFn
+			}
+			if tt.readMemStatsFn != nil {
+				memorylimiter.ReadMemStatsFn = tt.readMemStatsFn
```

Comment:
As you monkey patch `readMemStatsFn`, can we reuse the previous tests now instead of replacing them?

Reply:
ok ill put them back

Reply:
i could not consistently run the previous tests. I would still significantly modify the tests. is that still preferred?

Reply:
How significantly modify? Does exporting `CheckMemLimits` help?

Reply:
i pushed the mod test, it still tests all the conditions the previous tests did. it was just a lot of startup,shutdown,new processor. Since i cant control when the ml checkLimits() waits are there to ensure it ran after startup before consume

Reply:
`time.Sleep(10 * time.Millisecond)` can be flaky. Can we expose `CheckLimits` in the internal package and call it by the processor test?

Reply:
exposed CheckLimits, removed the sleeps